app/models/calculator_resolver.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class CalculatorResolver
+  def self.call(calculator)
+    dependencies = {}
+    get_calculations(calculator).each do |field|
+      dependencies[field] = if field[:type] == 'Calculation'
+                              get_dependencies(field[:value])
+                            else
+                              []
+                            end
```

Comment:
Use guard clause

Reply:
Thank you for the advice. I've reworked this chunk of code into guard clause.